src/main/java/fact/utils/SamplePedestalEvent.java

```diff
@@ -0,0 +1,350 @@
+/**
+ *
+ */
+package fact.utils;
+
+import fact.Utils;
+import fact.container.PreviousEventInfoContainer;
+import fact.io.hdureader.FITSStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import stream.Data;
+import stream.ProcessContext;
+import stream.StatefulProcessor;
+import stream.annotations.Parameter;
+import stream.expressions.version2.Condition;
+import stream.expressions.version2.ConditionFactory;
+import stream.io.JSONStream;
+import stream.io.SourceURL;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import static java.lang.Math.ceil;
+
+/**
+ * This processors changes the order of the pixels in the data from SoftId to Chid
+ *
+ * @author mbulinski
+ *
+ */
+public class SamplePedestalEvent implements StatefulProcessor {
+    static Logger log = LoggerFactory.getLogger(SamplePedestalEvent.class);
+
+    @Parameter(required = false, defaultValue="0", description = "The seed to make the results reproductable if desired.")
+    private long seed;
+
+    @Parameter(required = true, description = "The file containing the noise database for the noise to be used.")
+    private SourceURL noiseDatabase;
+
+    @Parameter(required = true, description = "The folder containing the data files.")
+    private String dataFolder;
+
+    @Parameter(required = false, defaultValue = "", description = "The value to prepend all keys from the noise file." +
+                    " (Exp: prependKey='Noise_', 'ZdPointing'->'Noise_ZdPointing'")
+    private String prependKey = "";
+
+    @Parameter(required = false, description = "The condition for the noise which will be used")
+    private String noiseCondition ="";
+    private Condition condition;
+
+    @Parameter(required = true, description="The binning of the zenith-angle. If single value: steplenght, bins into the given steplength"+
+                                            "If multivalue: Each value is the length of a single step starting with zenith=0.")
+    private String[] binning;
+
+    @Parameter(required = true, description = "The binning key of the double value from the noise database.")
+    private String dbBinningKey;
+
+    @Parameter(required = true, description = "The binning key of the double value from the item.")
```

Comment:
is this the key of the Zd value in the file of the event?

Reply:
please add to description: "e.g. key of the zd angle of the provided mc event. The value will be compared to the value of the dbBinningKey "